mr/src/main/java/org/elasticsearch/hadoop/serialization/bulk/AbstractBulkFactory.java

```diff
@@ -120,6 +121,12 @@ void doWrite(Object value) {
 
                 pool.get().bytes(valueString);
             }
+            
+            else if (value instanceof Date) {
+                String valueString = (value == null ? "null": Long.toString(((Date) value).getTime()));
+                pool.get().bytes(valueString);
+            }
```

Comment:
I agree that some refactoring is needed, but this small patch is what we applied in-house to be able to use the connector (we almost always use scripted updates, and almost always have datetime fields), so I'd appreciate if you can merge it as an interim fix before addressing the larger issue which may take longer.